app/services/affiliate-slots.js

```diff
@@ -123,10 +124,28 @@ export default Service.extend({
   logger: service(),
   geo: service(),
   wikiVariables: service(),
+  currentUser: service(),
 
   currentWikiId: readOnly('wikiVariables.id'),
   currentVertical: readOnly('wikiVariables.vertical'),
   currentCountry: readOnly('geo.country'),
+  currentUserId: readOnly('currentUser.userId'),
+
+  _updateUnitLink(unit, pageId = 'search') {
+    if (!unit || unit.campaign !== 'ddb') {
+      return unit;
+    }
+
+    const beaconId = Cookies.get('wikia_beacon_id');
+    const session = Cookies.get('wikia_session_id');
+    const userId = this.currentUserId ? this.currentUserId : 'null'; // make sure we actually send null
```

Comment:
We usually use 0 (or negative values) to denote anons

Reply:
I see. Good call